packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -0,0 +1,492 @@
+      );
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+    _creatingCompleter.complete();
+    return _creatingCompleter.future;
+  }
+
+  /// Prepare the capture session for video recording.
+  ///
+  /// Use of this method is optional, but it may be called for performance
+  /// reasons on iOS.
+  ///
+  /// Preparing audio can cause a minor delay in the CameraPreview view on iOS.
+  /// If video recording is intended, calling this early eliminates this delay
+  /// that would otherwise be experienced when video recording is started.
+  /// This operation is a no-op on Android.
+  ///
+  /// Throws a [CameraException] if the prepare fails.
+  Future<void> prepareForVideoRecording() async {
+    await CameraPlatform.instance.prepareForVideoRecording();
+  }
+
+  /// Captures an image and saves it to [path].
+  ///
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// If a file already exists at the provided path an error will be thrown.
+  /// The file can be read as this function returns.
+  ///
+  /// Throws a [CameraException] if the capture fails.
+  Future<XFile> takePicture() async {
+    if (!value.isInitialized || _isDisposed) {
+      throw CameraException(
+        'Uninitialized CameraController.',
+        'takePicture was called on uninitialized CameraController',
+      );
+    }
+    if (value.isTakingPicture) {
+      throw CameraException(
+        'Previous capture has not returned yet.',
+        'takePicture was called before the previous capture returned.',
+      );
+    }
+    try {
+      value = value.copyWith(isTakingPicture: true);
+      XFile file = await CameraPlatform.instance.takePicture(_cameraId);
+      value = value.copyWith(isTakingPicture: false);
+      return file;
+    } on PlatformException catch (e) {
+      value = value.copyWith(isTakingPicture: false);
+      throw CameraException(e.code, e.message);
+    }
+  }
+
+  /// Start streaming images from platform camera.
+  ///
+  /// Settings for capturing images on iOS and Android is set to always use the
+  /// latest image available from the camera and will drop all other images.
+  ///
+  /// When running continuously with [CameraPreview] widget, this function runs
+  /// best with [ResolutionPreset.low]. Running on [ResolutionPreset.high] can
+  /// have significant frame rate drops for [CameraPreview] on lower end
+  /// devices.
+  ///
+  /// Throws a [CameraException] if image streaming or video recording has
+  /// already started.
+  ///
+  /// The `startImageStream` method is only available on Android and iOS (other
+  /// platforms won't be supported in current setup).
+  ///
+  // TODO(bmparr): Add settings for resolution and fps.
+  Future<void> startImageStream(onLatestImageAvailable onAvailable) async {
+    assert(!Platform.isIOS && !Platform.isAndroid);
```

Comment:
Please, use `defaultTargetPlatform` from Flutter instead of `Platform` from `dart:io` to do these checks (there's one here, and another one in the `stopImageStream` method)

Reply:
Done!

Reply:
Thanks!!